sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala

```diff
@@ -339,7 +339,8 @@ case class DataSource(
         val baseRelation =
           dataSource.createRelation(sparkSession.sqlContext, caseInsensitiveOptions)
         if (baseRelation.schema != schema) {
-          throw new AnalysisException(s"$className does not allow user-specified schemas.")
+          throw new AnalysisException(s"$className does not allow user-specified schemas, " +
+              s"source schema: ${baseRelation.schema}, user-specific schema: ${schema}")
```

Comment:
Thank you for your contribution, @LiShuMing . Please make a unit test case for your contribution.
Otherwise, your contribution might be removed accidentally.

Reply:
OK~